generate/generate.py

```diff
@@ -697,8 +720,17 @@ def main():
             generate_all(discdoc)
         except Exception as e:
             print("Error while processing discovery doc")
-            raise e
+            # raise e
+            continue
+
+    _, _, mods_name = next(walk('../src/gen'))
+    mod_rs_file = open('../src/gen/mod.rs', 'w+')
+    for mod in mods_name:
+        if mod == 'mod.rs' or re.match(".*\.rs", mod) == None:
             continue
+        mod_name_line = 'pub mod ' + mod[:-3] + ';\n'
+        mod_rs_file.write(mod_name_line)
+    mod_rs_file.close()
 
```

Comment:
from what I can see, those are good changes

Reply:
I think this script can generate all the apis now without errors. It also automatically generates gen sub module (`src/gen/mod.rs`) for generated code's files.

Reply:
> > Thanks for your quick action.
> > I unified the common crate with the generated codes. Generated codes are now sub modules and reside in `src/gen`. It helps users to easily generate their desired apis and import only the main `async-google-apis` crate to use apis, it keeps the repo clean and straightforward. It has potential to use features macro and all supported generated code to publish the crate in crates.io.
> 
> About the first point: I think this is already the case; you generate one file for one API, and include the (MIT-licensed) code in your project. The a-g-a-c crate is used anyway, but it is small and not very complicated (no features, not many dependencies etc.)
> 
> Your approach seems to assume that a) the user clones and modifies the crate, locally or on crates.io or b) we publish authoritative API clients en masse, either in that crate or as many individual crates. It sounds like you had both cases in mind? Or am I missing the point?
> 
> Option (a) seems inconvenient, and option (b) is not very desirable for me, as I am not affiliated with Google (anymore), and honestly don't want to be on the hook for some user screaming "omg adwords-v1 API doesn't work anymore in Rust fixit fixit fixit!!1", when I never cared even a little bit about that particular API. By publishing the generator, users are encouraged to send me improvements to the generator instead :-)
> 
> About the second point: that is what I wanted to avoid. I used to collaborate a bit on Byron's google-apis-rs (didn't link him because it's only tangential), and he publishes crates for every API. This approach is alright, but I didn't want to do this mainly for the reason of not being too sure about the generated code quality (because the Discovery API is quite vast, I didn't make the effort to cover all possible cases; many APIs work fine, but it is not guaranteed that any new API will work well). My goal was rather to have a simple generator that you use rarely, with the generated code being checked in.
> 
> Please let me know if there is some misunderstanding, or if we simply disagree on the right approach to take. Like I said, I don't entirely hate your changes, it is just not clear yet how this would improve the situation for both the project (and me), and users.
> 
> Btw, if you or anyone else needs support (like in a commercial manner) for this, I'm definitely open to that. Just not necessarily in my free time 😉



> > Thanks for your quick action.
> > I unified the common crate with the generated codes. Generated codes are now sub modules and reside in `src/gen`. It helps users to easily generate their desired apis and import only the main `async-google-apis` crate to use apis, it keeps the repo clean and straightforward. It has potential to use features macro and all supported generated code to publish the crate in crates.io.
> 
> About the first point: I think this is already the case; you generate one file for one API, and include the (MIT-licensed) code in your project. The a-g-a-c crate is used anyway, but it is small and not very complicated (no features, not many dependencies etc.)
> 
> Your approach seems to assume that a) the user clones and modifies the crate, locally or on crates.io or b) we publish authoritative API clients en masse, either in that crate or as many individual crates. It sounds like you had both cases in mind? Or am I missing the point?
> 
> Option (a) seems inconvenient, and option (b) is not very desirable for me, as I am not affiliated with Google (anymore), and honestly don't want to be on the hook for some user screaming "omg adwords-v1 API doesn't work anymore in Rust fixit fixit fixit!!1", when I never cared even a little bit about that particular API. By publishing the generator, users are encouraged to send me improvements to the generator instead :-)
> 
> About the second point: that is what I wanted to avoid. I used to collaborate a bit on Byron's google-apis-rs (didn't link him because it's only tangential), and he publishes crates for every API. This approach is alright, but I didn't want to do this mainly for the reason of not being too sure about the generated code quality (because the Discovery API is quite vast, I didn't make the effort to cover all possible cases; many APIs work fine, but it is not guaranteed that any new API will work well). My goal was rather to have a simple generator that you use rarely, with the generated code being checked in.
> 
> Please let me know if there is some misunderstanding, or if we simply disagree on the right approach to take. Like I said, I don't entirely hate your changes, it is just not clear yet how this would improve the situation for both the project (and me), and users.
> 
> Btw, if you or anyone else needs support (like in a commercial manner) for this, I'm definitely open to that. Just not necessarily in my free time 😉

Publishing the crate was just an idea or the project potential and I'm not affiliated with the Google too, it's your decision 😉 
I've just tried to simplify the whole project structure, generating process and using the apis for ease of use.
In the current situation, users should add a-g-a-c dep along with generated code's deps and also moving around the generated code in proper directory in their project, but in my version they are all just one crate, only run generator and add the a-g-c dep and use the apis! want to updating the generated codes? only run the generator... (and don't need to moving around the files.)